### ISSUE TITLE
[Feature Request] Add an optional context menu prompt to enter an image filename before upload 

### ISSUE THREAD

**Issue by pranavAbe** [gavvvr/obsidian-imgur-plugin#53]
It would be beneficial, for organisational purposes, to have an optional toggle in the plugin settings, that provides a prompt to the user upon pasting in an image to enter a filename for the image before uploading, which also reflects within the Imgur dashboard. 

Support for other supported attributes by Imgur like Tags and Albums could also be considered an additional scope of improvement to this feature request.

**Comment by gavvvr** [comment]
Hi @pranavAbe. I agree. I initially wanted to upload images to specific album. I do not remember exactly why I did not implement it.. there was some impediment

**Comment by pranavAbe** [comment]
Well, is there a chance that I can ask you to look into this again? Perhaps also in terms of specifying a filename for upload and adding tags?

**Comment by gavvvr** [comment]
@pranavAbe , sure, you already did a request and it's accepted as a good idea (albums)
What do you mean by file name? Is it a name visible in your files list on imgur.com?

**Comment by pranavAbe** [comment]
Yeah, thanks for considering that. And there is a filename on Imgur. Right now the site is over capacity so I can't provide an example from the website, but here's what it looks like in the app. 

**Note:** The filename "Framework" is seen at the top. 

![The filename "Framework" is seen at the top](https://github.com/gavvvr/obsidian-imgur-plugin/assets/29583966/bf244e1f-3607-40e4-afa9-f45feb859e1f)

**Comment by gavvvr** [comment]
@pranavAbe I see now. The images uploaded by plugin are named "unknown". Do you know if it's also possible to rename already uploaded images?

**Comment by pranavAbe** [comment]
Yeah, it is possible on the website, although I haven't explored the API enough to know if there's an endpoint for the same on there.
